### PR TITLE
Export bugs

### LINKED ISSUE #4897: Minor bugs in new export tasks
**Describe the bug**
Test the new export tasks thoroughly and fix all issues. Some of the issues that I found already - 
null checks for nested table queries, etc

**To Reproduce**
1. Try the export as ical, xcal, pentabarfxml endpoitns.

**Expected behavior**
It should work without any problems.

**Additional details (please complete the following information):**
 - OS: Ubuntu 18.04

### PR DESCRIPTION
**Pull request by mayank8318**
Fixes #4897 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Test the export tasks and resolve minor bugs

#### Changes proposed in this pull request:
- Check for null microlocation before using microlocation.name
- Convert datetime to str for json serialization
- Correct event_detail endpoint url

### CODE REVIEW

File: app/api/helpers/pentabarfxml.py [hunk 1]
```diff
@@ -44,39 +44,41 @@ def export(event_id):
                                      .filter_by(state='accepted')
                                      .filter(Session.deleted_at.is_(None))
                                      .order_by(asc(Session.microlocation_id)).distinct())
+
             for microlocation_id in microlocation_ids:
```
Comment (srv-twry) [inline app/api/helpers/pentabarfxml.py]: May be you can rename this: ```microlocation_id``` to something like ```microlocation_array``` or `microlocation_tuple` etc depending on what it is. 
naming it `microlocation_id` and then reassigning it to `microlocation_id[0]` doesn't look good in terms of readability.

**Comment by mayank8318** [comment]
@srv-twry @bhaveshAn @Kreijstal @schedutron Please review

**Comment by mayank8318** [comment]
@srv-twry I hadn't squashed because each commit solved an independent issue. Should I still squash it?

**Comment by srv-twry** [comment]
@mayank8318 Oh, apologies. Please don't. It's good that each commit solves one bug. :+1:

**Comment by mayank8318** [comment]
@srv-twry Rebasing dismissed your review. Please review again :)

**Comment by mayank8318** [comment]
@bhaveshAn @Kreijstal @schedutron @pradeepgangwar Please review :)

**Pull request merged**